src/main/java/FrameworksDrivers/OtherAccount.java

```diff
@@ -60,49 +77,161 @@ public OtherAccount(JPanel masterPanel, CardLayout layout) {
+            label.setSize(100, 50);
+            this.coursesPanel.add(label.getLabel());
+        }
+
+        coursesHolder.add(coursesTitle.getLabel());
+        coursesHolder.add(this.coursesPanel);
+
+
         JPanel buttonAlign = new JPanel();
-        buttonAlign.setPreferredSize(new Dimension(600, 20));
-        buttonAlign.setLayout(new BoxLayout(buttonAlign, BoxLayout.X_AXIS));
-        backToMain = new Button();
-        backToMain.createButton(buttonAlign, "Back", 20, 20, 10, 10);
+        buttonAlign.setLayout(new FlowLayout(FlowLayout.LEFT));
+        buttonAlign.setPreferredSize(new Dimension(600, 35));
+        Button backToMain = new Button();
+        backToMain.createButton(buttonAlign, "Back", 30, 30, 30, 30);
         backToMain.getButton().addActionListener(this);
-        backToMain.setHorizontalAlignment("left");
+
 
         this.panel.add(buttonAlign);
         this.panel.add(inner);
         inner.add(upper);
         inner.add(lower);
+        this.panel.add(interestsHolder);
+        this.panel.add(attrHolder);
+        this.panel.add(coursesHolder);
 
+        JScrollPane scroller = new JScrollPane(this.panel,
+                JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+                JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+        this.panel.setLayout(new BoxLayout(this.panel, BoxLayout.Y_AXIS));
 
-        this.masterPanel.add(this.panel, "otherAccount");
-    }
+        this.masterPanel.add(scroller, "otherAccount");
 
+    }
 
+    /**
+     * Updates
+     *
+     * @parem paths References to pages that OtherAccount may switch to
+     */
     public void sendPaths(Object[] paths) {
         //gets paths to other pages
         this.paths = paths;
     }
 
     @Override
     public void actionPerformed(ActionEvent e) {
-        if(e.getSource() == this.backToMain.getButton()){
-            OtherAccountPresenter otherAccountPresenter = new OtherAccountPresenter(this.name.getLabel().getText());
-            this.layout.show(this.masterPanel, "mainpageView");
+        if (prevpage.equals("main")) {
+
+        } else {
+
         }
-    }
 
-    public void updatePage(Object[] info){
+    }
 
+    /**
+     * Gets and updates OtherAccount page with new user information when user switches to it
+     *
+     * @param info Object array that contains all the needed information to update OtherAccount
+     */
+    public void updatePage(Object[] info) {
         this.name.getLabel().setText((String) info[0]);
-        this.bio.getLabel().setText((String) info[1]);
-       // this.bio.getLabel().setText("<html>" + (String) info[2] + "</html>");
+        this.location.getLabel().setText((String) info[1]);
+        this.bio.getLabel().setText("<html>" + info[2] + "</html>");
 
-        //updates panel VVVVV
-        this.panel.revalidate();
-    }
+        ArrayList<String> intersts = (ArrayList<String>) info[3];
+        ArrayList<String> attributes = (ArrayList<String>) info[4];
```

Comment:
if these are instance variables, it would be more consistent to declare them nearer to the start of the file (it makes no difference, its just convention)